hugegraph-server/hugegraph-api/src/main/java/org/apache/hugegraph/api/traversers/KneighborAPI.java

```diff
@@ -179,18 +180,12 @@ public String post(@Context GraphManager manager,
         if (request.withVertex && !vertexIds.isEmpty()) {
             iterVertex = g.vertices(vertexIds.toArray());
             measure.addIterCount(vertexIds.size(), 0L);
-        } else {
-            iterVertex = vertexIds.iterator();
         }
 
-        Iterator<?> iterEdge = Collections.emptyIterator();
-        if (request.withPath) {
+        Iterator<Edge> iterEdge = Collections.emptyIterator();
+        if (request.withPath && request.withEdge) {
```

Comment:
looks like we cut the returning ids branch?
due to it is not easy for the client to determine whether it is an edge id or an edge object?